Framework/Algorithms/src/WienerSmooth.cpp

```diff
@@ -365,13 +366,16 @@ WienerSmooth::smoothSingleSpectrum(API::MatrixWorkspace_sptr inputWS,
 
   // copy the x-values and errors from the original spectrum
   // remove the last values for odd-sized inputs
+
+  out->setSharedX(0, inputWS->sharedX(wsIndex));
+  out->setSharedE(0, inputWS->sharedE(wsIndex));
```

Comment:
Keep this inside the `if` statement? The overhead may be very small, but it would also be clearer what the intention is. Also I am not sure if this would actually work. It might throw a size mismatch error for `ifOddSize == true`.